README.md

```diff
@@ -1,40 +1,33 @@
-# Deep Learning Model Extension Specification
+# ML Model Extension Specification
 
 [![hackmd-github-sync-badge](https://hackmd.io/3fB1lrHSTcSHQS57UhVk-Q/badge)](https://hackmd.io/3fB1lrHSTcSHQS57UhVk-Q)
 
-- **Title:** Deep Learning Model Extension
-- **Identifier:** <https://schemas.stacspec.org/v1.0.0-beta.3/extensions/dl-model/json-schema/schema.json>
-- **Field Name Prefix:** dlm
+- **Title:** Model Extension
+- **Identifier:** [https://schemas.stacspec.org/v1.0.0-beta.3/extensions/ml-model/json-schema/schema.json](https://schemas.stacspec.org/v1.0.0-beta.3/extensions/ml-model/json-schema/schema.json)
+- **Field Name Prefix:** mlm
 - **Scope:** Item, Collection
-- **Extension [Maturity Classification][stac-ext-maturity]:** Proposal
-- **Owner**:
-  [@sfoucher](https://github.com/sfoucher)
-  [@fmigneault](https://github.com/fmigneault)
-  [@ymoisan](https://github.com/ymoisan)
-
-[stac-ext-maturity]: https://github.com/radiantearth/stac-spec/tree/master/extensions/README.md#extension-maturity
-
-This document explains the Template Extension to the [SpatioTemporal Asset Catalog][stac-spec] (STAC) specification.
-This document explains the fields of the STAC Deep Learning Model (dlm) Extension to a STAC Item. 
-The main objective is to be able to build model collections that can be searched 
-and that contain enough information to be able to deploy an inference service.
-When Deep Learning models are trained using satellite imagery, it is important
-to track essential information if you want to make them searchable and reusable:
-1. Input data origin and specifications
-2. Model base transforms
-3. Model output and its semantic interpretation
-4. Runtime environment to be able to run the model
+- **Extension Maturity Classification:** Proposal
+- **Owner:**
+  - [@sfoucher](https://github.com/sfoucher)
+  - [@fmigneault](https://github.com/fmigneault)
+  - [@ymoisan](https://github.com/ymoisan)
+  - [@rbavery](https://github.com/rbavery)
+
+This document explains the Template Extension to the [SpatioTemporal Asset Catalog (STAC)](https://github.com/radiantearth/stac-spec) specification. This document explains the fields of the STAC Model Extension to a STAC Item. The main objective of the extension is two-fold: 1) to enable building model collections that can be searched alongside associated STAC datasets and 2) to record all necessary parameters, artifact locations, and high level processing steps to deploy an inference service. Specifically, this extension records the following info to make ML models searchable and reusable:
+1. Sensor band specifications
+2. The two fundamental transforms on model inputs: rescale and normalization
+3. Model output shape, data type, and its semantic interpretation
+4. An optional, flexible description of the runtime environment to be able to run the model
```

Comment:
My thinking here is that requiring the runtime will be too big a barrier for most folks looking to publish a model. As it becomes easier to distribute [AOT compiled models](https://pytorch.org/docs/stable/export.html) instead of model weights, I think it is also becoming more feasible to simply specify the torch version required to load a model and run inference. 

Ultimately I want to nudge most publishers of models to only require users to

```
# torch version is described by the model metadata
import torch
# no need for complex dependency management to load the model architecture seaparate from the model weights/state
model = torch.load(model_path)
# expected_input is described by the model metadata
results = model(expected_input)
```

but want to offer the option to provide a runtime description for models that still require complex dependencies.

I realize this is a big change from the original direction and that while lots of folks are using torch 2.0 and other frameworks that support AOT compiled models, not everyone uses this functionality or even knows it exists. Very open to discussing the optionality more. My hope is that this extension can help nudge people toward best practices when distributing models for inference (AOT compilation for portability and performance).

Reply:
I think that even with this very minimalistic definition as above, a runtime would be needed at the very least to provide a docker reference that has an appropriate pytorch version installed, or alternatively/additionally, fields such as `framework` and `version` to indicate for example if pytorch 2.0 is required. The `model_asset` field would most realistically refer to the `model_path` to retrieve for this script, and this script itself could be referred to by `source_code_url` to provide instructions on how to run the model. 

I agree that most of the other runtime parameters should be optional to lower the adoption barrier. However, these should at least be provided to ensure some minimal form of reproducible experiments or references to figure out how to instantiate the model properly.

Reply:
I think we're mostly in agreement here. I lean more toward making framework and version required, and leaving the container reference optional (which might be docker or another container system). 

I think enforcing the structure of the container file is outside the scope of the extension. I worry that requiring such a flexible requirement might confuse or put off some folks who don't typically containerize their projects, and who would otherwise use the extension to document their model. In particular, model publishers in academia tend to be less avid users of Docker compared to industry. 

When working with a new model, my gold standard is definitely a dockerfile, but I'd be happy to encounter a github repo with just a clear description of model input requirements, a clear description of the output structure, and where to acquire the model. Then I typically would make my own Dockerfile to run the model if needed.

Even if the model publisher specified a Dockerfile to run the model, I'd probably usually want to change the base image and other aspects of it for my own use case.

I think this merits more discussion so happy to move forward with whatever approach you think is best but then seek feedback in an issue and revisit if needed. 

In any case we can suggest popular base images like torchserve and tfserving or the pytorch and tensorflow base images to give folks an entrypoint to containerize.

Reply:
an example of aversion to docker/.devcontainers: https://github.com/microsoft/torchgeo/pull/1085#issuecomment-1416381540

many models on huggingface related to geospatial or other models don't have associated Dockerfiles.

Reply:
I agree with the points you mentioned.
I also had in mind that the example definition would point at an official pytorch image to guide users with the most common use case.

Reply:
sounds good I added examples for the official base images for pytorch and tf and both frameworks' serving options